numpy/core/defchararray.py

```diff
@@ -1398,6 +1400,147 @@ def rstrip(a, chars=None):
+    to catch under normal circumstances.
+
+    Parameters
+    ----------
+    a : array-like
+        Must have dtype `numpy.str_` or `numpy.unicode_`.
+    start : int
+        The start of the slice. Treated as `stop` if `stop` is not
+        provided.
+    stop : int, optional
+        The end of the slice.
+    step : int, optional
+        The number of steps between characters. Usually one, but can be
+        any other step size.
+    chunksize : int
+        The number of characters to extract at each step. Usually one,
+        but can be any other amount that fits into the length of the
+        string. Must be a positive integer.
+
+    Return
+    ------
+    slice : np.ndarray
+        A view of the original data, sliced to show strings of the
+        required size. The dimensions of the new array are the same
+        as those of the input for `step == chunksize == 1`, and
+        ``(*a.shape, (stop - start) // step)`` (after bounds are
+        computed) otherwise. The datatype is the same as that of `a`,
+        with same byte order and code as the input. Only the element
+        length differs, being either `stop - start` or `chunksize`,
+        respectively.
+    """
+    a = asanyarray(a)
+    dtype = a.dtype
+
+    if dtype.char not in 'US':
+        raise TypeError('Only U and S string datatypes supported. '
+                        f'Found {dtype.char}')
+
+    if chunksize is None:
+        chunksize = 1
+    elif chunksize < 1:
+        raise ValueError('chunksize must be a positive integer')
+
+    length = int(dtype.str[2:])
+
+    # Adjust the bounds using a slice object
+    if stop is None and step is None:
+        start, stop = 0, start
+    start, stop, step = slice(
+                start, stop, step
+            ).indices(max(0, length - chunksize + 1))
+    if start < 0:
+        start = stop = 0
+        step = 1
+
+    # Get the real dtype information
+    charsize = dtype.itemsize // length
+
+    # Find the real base array
+    base = a
+    while base.base is not None:
+        base = base.base
+    # ['data'][0] is the memory pointer to the buffer
+    realoffset = (a.__array_interface__['data'][0] -
+                  base.__array_interface__['data'][0])
+
+    # Compute the final view parameters
+    if step == 1 and chunksize == 1:
+        chunksize = max(stop - start, 0)
+        newshape = a.shape
+        newstrides = a.strides
+    else:
+        newshape = (*a.shape, max(0, (stop - start + (step - numpy.sign(step))) // step))
+        newstrides = (*a.strides, step * charsize)
+    newdtype = numpy.dtype(f'{dtype.str[:2]}{chunksize}')
+    newoffset = start * charsize
+
+    try:
+        # This should work 99% of the time
+        newarray = ndarray(buffer=base, offset=newoffset + realoffset,
+                           shape=newshape, strides=newstrides,
+                           dtype=newdtype)
+    except ValueError as e:
```

Comment:
It should be possible to achieve this with `stride_tricks` so that a copy is never returned.

Reply:
Good point. Let me see what stride tricks is doing.

Reply:
@eric-wieser. I have to ask. How is [this line](https://github.com/numpy/numpy/blob/f224ca3c43260730d43636a42d4670735f144906/numpy/lib/stride_tricks.py#L104), in `as_strided` fundamentally different from what you marked above: `array = np.asarray(DummyArray(interface, base=x))`. Is is because `DummyArray` is "trusted" because it's already an array subclass, vs what I have here, which is attempting to make a new one from scratch? The only thing that won't work here is that `as_strided` does not accept an offset. I can hack in another parameter to do that.

Reply:
@eric-wieser I tried it your way: https://github.com/madphysicist/numpy/commit/e42218658aeb35b236226bcdeac11fc9c6a7d3b1. However, as it turns out, it makes no difference. The issue is with changing dtype: both `ndarray` and `as_strided` complain equally when you try to do that unless the array is contiguous. Maybe that's something worth bothering about, maybe not. Within the scope of this PR, doing what I'm doing now seems to be a fix. I do need to add a test for an array created as a non-contiguous view into a `bytes` object or something.

Reply:
@eric-wieser Please take a look at the changes and additional tests I made to deal with non-contiguous base arrays. I basically made the assumption that somewhere there is a contiguous block of memory available, got its address and calculated the size out to the last element I need, then used that to create my view. This is a deficiency in the API: I left a ranty comment about it in the code. There is no reason a responsible user shouldn't be able to use `as_strided` to view with a new dtype. In the meantime, please enjoy my hack.

Reply:
@eric-wieser Given that I basically agree with your preference for using `as_strided`, would you prefer that I move the dtype-modifying code out to `as_strided`, similarly to the private branch I linked above? It would definitely make this function cleaner and less error prone.